R/doubletFinder.R

```diff
@@ -87,13 +97,151 @@ doubletFinder <- function(seu, expected.doublets = 0, proportion.artificial = 0.
   }
 
   ## Step 7: Add results to metadata slot of original Seurat object
-  seu <- Seurat::AddMetaData(seu, metadata = pANN, col.name = "pANN")
+  obj <- Seurat::AddMetaData(obj, metadata = pANN, col.name = "pANN")
   predictions <- as.data.frame(rep("Singlet", n_real.cells), ncol = 1, stringsAsFactors = FALSE)
   rownames(predictions) <- real.cells
-  doublet.predictions <- rownames(seu@meta.data)[order(seu@meta.data$pANN, decreasing = TRUE)]
+  doublet.predictions <- rownames(obj@meta.data)[order(obj@meta.data$pANN, decreasing = TRUE)]
   doublet.predictions <- doublet.predictions[1:expected.doublets]
   predictions[doublet.predictions, ] <- "Doublet"
   colnames(predictions) <- "pANNPredictions"
-  seu <- Seurat::AddMetaData(seu, metadata = predictions, col.name = "pANNPredictions")
-  return(seu)
+  obj <- Seurat::AddMetaData(obj, metadata = predictions, col.name = "pANNPredictions")
+  return(obj)
+}
+
+#' @param gene.names Dataset name in loom object for gene names; duplicates are not allowed.
```

Comment:
Unless we would like to implement a version where you run doubletFinder on only a subset of cells/genes, I think these arguments are unnecessary. Making artificial doublets using var.genes could be an interesting way to speed up the process, although the gains will likely be minimal because the main limiting factor is distance matrix computation. It could also be useful to generate doublets only using cells run on the same 10X/Drop-Seq lane, but that would be an internal modification incorporating new metadata instead of an external cell subsetting.

Reply:
I don't mean to use a subset of cells/genes in those arguments. Seurat defaults to "col_attrs/cell_names" for cell names and "row_attrs/gene_names" to store gene names. However, I got a dataset that stores cell names and gene names under column or row attributes with different names, such as "col_attrs/CellID", and sometimes I prefer to use transcript accessions in place of gene names since there're different isoforms of the same gene and I don't want to call accessions "gene_names", so it's desirable to have the option not to stick to the names "col_attrs/cell_names" and "row_attrs/gene_names". That's not about a subset of cells/genes, but about the name of the datasets where cell names and gene names are stored.